stdlib/ephemeron.ml

```diff
@@ -21,7 +21,24 @@ module type SeededS = sig
 end
 
 module type S = sig
-  include Hashtbl.S
+  type key
+  type 'a t
+  val create : int -> 'a t
+  val clear : 'a t -> unit
+  val reset : 'a t -> unit
+  val copy : 'a t -> 'a t
+  val add : 'a t -> key -> 'a -> unit
+  val remove : 'a t -> key -> unit
+  val find : 'a t -> key -> 'a
+  val find_opt : 'a t -> key -> 'a option
+  val find_all : 'a t -> key -> 'a list
+  val replace : 'a t -> key -> 'a -> unit
+  val mem : 'a t -> key -> bool
+  val iter : (key -> 'a -> unit) -> 'a t -> unit
+  val filter_map_inplace: (key -> 'a -> 'a option) -> 'a t -> unit
+  val fold : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+  val length : 'a t -> int
+  val stats: 'a t -> Hashtbl.statistics
```

Comment:
I'm not even sure what it means to iterate on a weak hashtable… I will certainly not do that before #1010 is merged, my quota of motivation is running dry right now.